src/main/java/uk/co/ramp/covid/simulation/util/PopulationGenerator.java

```diff
@@ -8,15 +8,26 @@
     
     private static final int RETRIES = 20;
 
-    public static Population genValidPopulation(int populationsize) {
+    public static Population genValidPopulation(int populationsize) throws CannotGeneratePopulationException {
         Population p = null;
-        for (int i = 0; i < RETRIES; i++) {
+        for (int i = 0; i < RETRIES - 1; i++) {
             try {
                 p = new Population(populationsize);
+                break;
             } catch (ImpossibleAllocationException | ImpossibleWorkerDistributionException e2) {
                 continue;
```

Comment:
I think it would be good to write a warning (or 'info') to the log when a retry is necessary.